pkg/planner/core/find_best_task.go

```diff
@@ -3392,10 +3392,11 @@ func getOriginalPhysicalTableScan(ds *logicalop.DataSource, prop *property.Physi
 	ts.SetSchema(ds.Schema().Clone())
 	rowCount := path.CountAfterAccess
 	// Add an arbitrary tolerance factor to account for comparison with floating point
-	if (prop.ExpectedCnt + cost.ToleranceFactor) < ds.StatsInfo().RowCount {
+	if (prop.ExpectedCnt+cost.ToleranceFactor) < ds.StatsInfo().RowCount ||
+		(isMatchProp && min(ds.StatsInfo().RowCount, prop.ExpectedCnt) < rowCount && len(path.AccessConds) > 0) {
```

Comment:
```ds.StatsInfo().RowCount``` can be placed above the if statement.

It will look like this.
```
rowCount := ds.StatsInfo().RowCount

if (xxx < rowCount) || min(rowcount, prop.ExpectedCnt) {
```